components/lwip/port/esp32/include/arch/cc.h

```diff
@@ -35,7 +35,9 @@
 #define __ARCH_CC_H__
 
 #include <stdint.h>
+#ifndef LWIP_PROVIDE_ERRNO
 #include <errno.h>
+#endif
```

Comment:
Perhaps this include can probably be removed altogether, and instead we could define `LWIP_ERRNO_STDINCLUDE` in components/lwip/port/esp32/include/lwipopts.h? Then `<errno.h>` will be included automatically from components/lwip/lwip/src/include/lwip/errno.h.

Reply:
Yes, that should also fix it because `LWIP_PROVIDE_ERRNO` is checked first before including `<errno.h>`, so I'll amend this PR. That said, I noticed that some files include `<errno.h>` directly, I think they should include `lwip/errno.h`. For example `components/esp_http_server/src/httpd_main.c`

Reply:
It seems that I have to revert changes in my projects and use `<errno.h>` anyway, but I'll keep this PR open if you say it is useful. Btw would this `components/lwip/lwip/test/unit/cc_esp_platform.h` cause any unit tests to fail after this change ?